data-model/src/main/scala/za/co/absa/enceladus/model/conformanceRule/ConnectedEntity.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.model.conformanceRule
+
+sealed trait ConnectedEntity{
+  val name: String
+  val version: Int
+  val kind: String
```

Comment:
I think this is most usually dubbed as
```
`type`
``` 
(backticks needed as it is a reserved word, but I can live with `kind` as well  :) This approach is definitely an improvement over the previous state IMHO.

Reply:
Because `type` is a reserved word, I think `kind` is better here.